File: libgx_shark_adventure/android/src/com/nopalsoft/sharkadventure/AndroidLauncher.java
package com.nopalsoft.sharkadventure;

import android.os.Bundle;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.backends.android.AndroidApplication;
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;
import com.nopalsoft.sharkadventure.handlers.RequestHandler;

public class AndroidLauncher extends AndroidApplication implements RequestHandler {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AndroidApplicationConfiguration config = new AndroidApplicationConfiguration();
        initialize(new MainShark(this), config);
    }

    @Override
    public void showInterstitial() {

    }

    @Override
    public void shareOnTwitter(String mensaje) {
        String tweetUrl = "https://twitter.com/intent/tweet?text=" + mensaje + " Download it from &url=https://play.google.com/&hashtags=SharkAdventure";
        Gdx.net.openURI(tweetUrl);
    }

    @Override
    public void showAdBanner() {

    }

    @Override
    public void hideAdBanner() {

    }
}

File: libgx_shark_adventure/android/src/com/nopalsoft/sharkadventure/AndroidLauncher.kt
package com.nopalsoft.sharkadventure

import android.os.Bundle
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.backends.android.AndroidApplication
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration
import com.nopalsoft.sharkadventure.handlers.FacebookHandler
import com.nopalsoft.sharkadventure.handlers.RequestHandler

class AndroidLauncher : AndroidApplication(), RequestHandler, FacebookHandler {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val config = AndroidApplicationConfiguration()
        initialize(MainShark(this, this), config)
    }

    override fun facebookSignIn() {
    }

    override fun showFacebook() {
        Gdx.net.openURI("https://www.facebook.com")
    }


    override fun showInterstitial() {
    }

    override fun shareOnTwitter(message: String) {
        val tweetUrl = "https://twitter.com/intent/tweet?text=$message Download it from &url=https://play.google.com/&hashtags=SharkAdventure"
        Gdx.net.openURI(tweetUrl)
    }

    override fun showAdBanner() {
    }

    override fun hideAdBanner() {
    }
}
